### DISCUSSION TITLE
Testing F Prime Performance

### DISCUSSION THREAD

**Discussion by frankdwhitworth** [nasa/fprime#2859] [General]
Hello all - 

My name is Frank Whitworth and I am currently a flight software engineer, primarily working with Core Flight System (cFS), for NASA's Gateway spacecraft.
I am also a Graduate Student at Georgia Tech, working on research in the flight software field. 

Recently, there was a paper comparing and contrasting F Prime with cFS, and laying out "trade considerations" when deciding which framework to use.
https://digitalcommons.usu.edu/smallsat/2024/all2024/206/

I am embarking on some research work which will aim to contribute to this comparison from a performance point of view.
Since my actual work with the Gateway spacecraft is just in cFS, I am new to the F Prime framework.
So, I would love to have conversations about how to capture, characterize, and compare F Prime's performance.

Me and my team thought that this would be the best forum to discuss so that contributions might not be limited to "behind closed door" meetings, and everyone can benefit or contribute to this :)

Happy developing and looking forward to working with the F Prime community!

-Frank Whitworth
Flight Software Engineer, NASA
Graduate Student, Georgia Tech

**Comment by timcanham** [comment]
@frankdwhitworth If you have a way to get to JPL, we are holding a training workshop here in October: https://nasa.github.io/fprime/upcoming.html#flight-softwaref-developer-workshop-at-jpl

**Reply by frankdwhitworth** [reply]
@timcanham Thank you for your quick response! Unfortunately, I don't think I can make it to JPL in October for the training workshop

I was hoping for your thoughts on how you would characterize and measure F Prime's performance as a flight software framework. There are many ways one could measure F Prime's macro ability to ingest commands and produce command responses, or ingest telemetry requests and telemetry responses. However, I am looking more at measuring the framework itself and being able to evaluate those hard metrics and evidence.

It will be hard because the two frameworks are so different, but I figured I would ask about how to represent and measure F Prime's framework.

Hope you have a great weekend!

-Frank

**Reply by LeStarch** [reply]
What metrics are you looking at specifically?  It is hard to point you at how to do performance characterization without knowing what performance you are looking at.  This could be anything from developer efficiency, to realtime jitter.

**Reply by SterlingPeet** [reply]
I think this is actually the question being asked, as in "What might I look at to showcase F Prime performance?"

Or put a different way: what about using F Prime is measurably compelling?   e.g.: developer efficiency to realtime jitter?  If somebody asks _why_ F Prime should be used, what do you point at?
